python-checks/src/main/java/org/sonar/python/checks/BooleanCheckNotInvertedCheck.java

```diff
@@ -127,20 +127,22 @@ private static String getNewExpression(BinaryExpression toUse, String oppositeOp
     return getText(toUse.leftOperand()) + " " + oppositeOperator + " " + getText(toUse.rightOperand());
   }
 
-  private static String getText(Tree tree) {
-    return TreeUtils.tokens(tree).stream()
-      .map(Token::value)
-      .reduce("", (acc, currentChar) -> isSpaceNotNeeded(acc, currentChar) ? (acc + currentChar) : (acc + " " + currentChar));
-  }
-
-  @SuppressWarnings("java:S125")
-  private static boolean isSpaceNotNeeded(String acc, String toAdd) {
-    // Heuristic telling when a space is not needed: 1) after a space, after opening parenthesis or bracket;
-    // 2) if the accumulator is ending with anything other than a comma and that an opening parenthesis is to be added;
-    // 3) if the accumulator is ending with anything other than a space and that an opening bracket is to be added
-    // 4) if a specific character is added : such as closing parenthesis or bracket, or a comma.
-    return acc.isBlank() || acc.endsWith("(") || acc.endsWith("[")
-      || (!acc.endsWith(",") && "(".equals(toAdd)) || (!acc.endsWith(" ") && "[".equals(toAdd))
-      || "]".equals(toAdd) || ")".equals(toAdd) || ",".equals(toAdd);
+  private static String getText(Tree defaultValue) {
```

Comment:
For the time being is ok, however next time we need this we might think of moving this to a utility class.